agent.go

```diff
@@ -226,12 +227,22 @@ func (a *Agent) writeOutput(
 	start := time.Now()
 
 	for {
-		err := ro.Output.Write(points)
+		if (len(ro.Config.Pass) + len(ro.Config.Drop) + len(ro.Config.TagDrop) + len(ro.Config.TagPass)) > 0 {
```

Comment:
can you break out this logic into a separate function? Maybe make it a function of the `RunningOutput` struct that takes in points as an argument.